apps/web-app/components/settings/addNewGroup.vue

```diff
@@ -4,47 +4,264 @@
+          selectedDomain.value?.domainPublicId &&
+          newGroupNameValid.value === true &&
+          !!newGroupColorValue.value
+        );
+      }
+      return (
+        newIdentityUsernameValid.value === true &&
+        (newIdentitySendNameValid.value || newGroupNameValid.value) === true &&
+        selectedDomain.value?.domainPublicId &&
+        !!newGroupColorValue.value
+      );
+    }
     return newGroupNameValid.value === true && !!newGroupColorValue.value;
   });
 
+  const toggleableOptions = ref<{
+    address: boolean;
+    sendName: boolean;
+    domainName: boolean;
+  }>({
+    address: false,
+    sendName: false,
+    domainName: false
+  });
+
+  const toggleInputOption = (option: keyof typeof toggleableOptions.value) => {
+    toggleableOptions.value[option] = !toggleableOptions.value[option];
+    if (option === 'address' && !toggleableOptions.value.address) {
+      newIdentityUsernameValid.value = null;
+      newIdentityUsernameValidationMessage.value = '';
+    }
+    if (option === 'sendName' && !toggleableOptions.value.sendName) {
+      newIdentitySendNameValid.value = null;
+    }
+    if (option === 'domainName' && !toggleableOptions.value.domainName) {
+      newDomainNameValid.value = null;
+    }
+  };
+
   const emit = defineEmits(['close']);
 
   const route = useRoute();
 
   const orgSlug = route.params.orgSlug as string;
 
   async function createGroup() {
-    if (!newGroupColorValue.value) return;
-    buttonLoading.value = true;
-    buttonLabel.value = 'Creating...';
-    const { newGroupPublicId } =
-      await $trpc.org.users.userGroups.createOrgUserGroups.mutate({
-        groupName: newGroupNameValue.value,
-        groupDescription: newGroupDescriptionValue.value,
-        groupColor: newGroupColorValue.value
+    if (createEmailIdentityForGroup.value) {
+      if (!newGroupColorValue.value) return;
+      buttonLoading.value = true;
+      buttonLabel.value = 'Creating...';
+      const toast = useToast();
+      const { newGroupPublicId } =
+        await $trpc.org.users.userGroups.createOrgUserGroups.mutate({
+          groupName: newGroupNameValue.value,
+          groupDescription: newGroupDescriptionValue.value,
+          groupColor: newGroupColorValue.value
+        });
+
+      const selectedGroupsPublicIds: string[] = selectedOrgGroups.value.map(
+        (group) => group.publicId as string
+      );
+      const selectedOrgMembersPublicIds: string[] =
+        selectedOrgMembers.value.map((member) => member.publicId as string);
+
+      await $trpc.org.mail.emailIdentities.createNewEmailIdentity.mutate({
+        emailUsername: newIdentityUsernameValue.value,
+        domainPublicId: selectedDomain.value?.domainPublicId as string,
+        sendName: newIdentitySendNameValue.value,
+        routeToGroupsPublicIds: [...selectedGroupsPublicIds, newGroupPublicId],
+        routeToUsersOrgMemberPublicIds: selectedOrgMembersPublicIds,
+        catchAll: newIdentityCatchAll.value
```

Comment:
we can remove these two arrays
and just send 
` routeToGroupsPublicIds: [newGroupPublicId],
        routeToUsersOrgMemberPublicIds: [],`